test/test_scripts_run_task.py

```diff
@@ -270,3 +272,56 @@ def _Popen(
 
     assert os.path.isdir(untracked_dir.name) is False
     assert os.path.isfile(untracked_file.name) is False
+
+
+@pytest.fixture(scope="session")  # Tests shouldn't change this repo
+def mock_git_repo():
+    "Mock repository with files, commits and branches for using as source"
+    with tempfile.TemporaryDirectory() as repo:
+        repo_path = str(repo)
+        subprocess.check_call(["git", "init", "-b", "main"], cwd=repo_path)
+
+        def _commit_file(message):
+            untracked_file = tempfile.NamedTemporaryFile(dir=repo, delete=False)
+            untracked_file.write(b"test file content")
+            untracked_file.close()
+            subprocess.check_call(["git", "add", "."], cwd=repo_path)
+            subprocess.check_call(["git", "commit", "-m", message], cwd=repo_path)
+
+        _commit_file("Initial commit")
+        subprocess.check_call(["git", "checkout", "-b", "mybranch"], cwd=repo_path)
+        _commit_file("File in my branch")
+        subprocess.check_call(["git", "checkout", "main"], cwd=repo_path)
+        yield repo_path
+
+
+@pytest.mark.parametrize(
+    "base_ref,ref",
+    [
+        (None, None),
+        (None, "main"),
+        (None, "mybranch"),
+        ("main", "main"),
+        ("main", "mybranch"),
+    ],
+)
+def test_git_checkout(
+    mock_stdin,
+    run_task_mod,
+    mock_git_repo,
+    base_ref,
+    ref,
+):
+    with tempfile.TemporaryDirectory() as workdir:
+        destination = os.path.join(workdir, "destination")
+        run_task_mod.git_checkout(
+            destination_path=destination,
+            head_repo=mock_git_repo,
+            base_repo=mock_git_repo,
+            base_ref=base_ref,
+            base_rev=None,
+            ref=ref,
+            commit=None,
+            ssh_key_file=None,
+            ssh_known_hosts_file=None,
+        )
```

Comment:
Test: let's assert the repo is checked out to the actual revision we're expecting. This can be done by asserting whether a given file exists or now.